### PR TITLE
Fixing env loading issue

### PR DESCRIPTION
**Pull request by mhmdksh**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@ version: '3.9'
 services:
   givpower-bot:
     image: ghcr.io/giveth/givpower-bot:main
-    environment:
-      - NODE_ENV=develop
+    env_file:
+      - ./config/.env
```
Comment (aminlatifi) [inline docker-compose.yml:7]:
```suggestion
    environment:
      - NODE_ENV=production
```

File: config/.env.template [hunk 2]
```diff
@@ -3,6 +3,7 @@ POLL_PERIOD_SECOND=300
 SUBGRAPH_ENDPOINT='https://api.thegraph.com/subgraphs/name/aminlatifi/givpower-deployment-six'
 GIVPOWER_CONTRACT_ADDRESS='0x898Baa558A401e59Cb2aA77bb8b2D89978Cf506F'
 PRIVATE_KEY=''
-UNLOCK_PER_TRANSACTION = 10
-GAS_MAX_BASE_FEE: 1.5;
-GAS_PRIORITY_FEE: 1.5;
+UNLOCK_PER_TRANSACTION=10
+GAS_MAX_BASE_FEE=1.5
+GAS_PRIORITY_FEE=1.5
+NODE_ENV=develop
```
Comment (aminlatifi) [inline config/.env.template:9]: There is no need to set `NODE_ENV`  here, the env file is already chosen by value of this env var already

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: config/.env.template (hunk 2 above)
Comment (mhmdksh) [reply config/.env.template:9]: > There is no need to set `NODE_ENV` here, the env file is already chosen by value of this env var already

True, but the idea is that in this way, the docker-compose file can be as clean as possible and not dependent on what ENV values are put inside it. And we throw all the env configs to the .env file

File: docker-compose.yml (hunk 1 above)
Comment (mhmdksh) [reply docker-compose.yml:7]: Again, this will keep the dependency on the structure of the docker-compose file. Which is not advisable

**Pull request merged**